packages/apollo-gateway/src/index.ts

```diff
@@ -343,40 +287,87 @@ export class ApolloGateway implements GraphQLService {
     };
   }
 
-  private startPollingServices() {
-    if (this.pollingTimer) clearInterval(this.pollingTimer);
+  private async updateComposition(options?: {
+    engine?: GraphQLServiceEngineConfig;
+  }) {
+    const previousSchema = this.schema;
+    const previousServiceDefinitions = this.serviceDefinitions;
+    const previousCompositionMetadata = this.compositionMetadata;
 
-    this.pollingTimer = setInterval(async () => {
-      let result: Await<ReturnType<Experimental_UpdateServiceDefinitions>>;
-      try {
-        result = await this.updateServiceDefinitions(this.config);
-      } catch (e) {
-        this.logger.debug(
-          'Error checking for schema updates. Falling back to existing schema.',
-          e,
-        );
-        return;
-      }
+    if (options && options.engine) {
```

Comment:
We could probably take this bit out and put it in the constructor or in the `load`? It's modifying class-level variables and its referring to arguments that should not be passed in in a dynamic way. Unless we're trying to support auto-reload of changing our API key, but I think not (and this doesn't actually accomplish that)?